src/api_server/src/attest/amber.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023 by Intel.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::Attest;
+use anyhow::*;
+use as_types::AttestationResults;
+use async_trait::async_trait;
+use jsonwebtoken::{decode, decode_header, jwk, DecodingKey, Validation};
+use kbs_types::{Attestation, Tee};
+use reqwest::header::{ACCEPT, CONTENT_TYPE};
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::BufReader;
+
+#[derive(Deserialize, Debug)]
+struct TdxEvidence {
+    _cc_eventlog: Option<String>,
+    quote: String,
+}
+
+#[derive(Serialize, Debug)]
+struct AttestReqData {
+    quote: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct AttestRespData {
+    token: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Claims {
+    amber_unmatched_policy_ids: Option<Vec<serde_json::Value>>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct AmberConfig {
+    pub base_url: String,
+    pub api_key: String,
+    pub certs_file: String,
+    pub check_policy: Option<bool>,
+}
+
+pub struct Amber {
+    config: AmberConfig,
+    jwks: jwk::JwkSet,
+}
+
+#[async_trait]
+impl Attest for Amber {
+    async fn attest_verify(
+        &mut self,
+        tee: Tee,
+        _nonce: &str,
```

Comment:
You need to verify that the _nonce is part of the attestation evidence.

Reply:
The nonce should be verify, there is already a patch to verify nonce and tee_pubkey.
https://github.com/Lu-Biao/kbs/commit/db6cf8f28f422e8992b5575cbe05ea3268fc9558


but, amber require the report data should be SHA512, but current attestation- agent use SHA384


https://github.com/confidential-containers/attestation-agent/blob/2dd57060343b646167dd6d850fe19900e4ab32fc/kbs_protocol/src/lib.rs#L67


https://github.com/confidential-containers/attestation-agent/blob/2dd57060343b646167dd6d850fe19900e4ab32fc/deps/crypto/src/lib.rs#L96


After updating to SHA512,  I can submit the PR.